src/elements/PolygonAperture.H

```diff
@@ -0,0 +1,297 @@
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Eric G. Stern, Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_POLYGONAPERTURE_H
+#define IMPACTX_POLYGONAPERTURE_H
+
+
+#include "particles/ImpactXParticleContainer.H"
+#include "mixin/alignment.H"
+#include "mixin/beamoptic.H"
+#include "mixin/thin.H"
+#include "mixin/lineartransport.H"
+#include "mixin/named.H"
+#include "mixin/nofinalize.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_Math.H>
+#include <AMReX_REAL.H>
+#include <AMReX_SIMD.H>
+#include <ablastr/constant.H>
+#include <cmath>
+#include <stdexcept>
+
+
+namespace impactx::elements
+{
+
+    /** Dynamic data for the PolygonAperture elements
+     *
+     * Since we copy the element to the device, we cannot store this data on the element itself.
+     * But we can store pointers to this data with the element and keep a lookup table here,
+     * which we clean up in the end.
+     */
+    namespace PolygonApertureData
+    {
+        //! last used id for a created ExactMultipole
+        inline int next_id = 0;
+
+        //! host: normal multipole coefficients of the magnetic field
+        inline std::map<int, std::vector<amrex::ParticleReal>> h_vertices_x = {};
+        //! host: skew multipole coefficients of the magnetic field
+        inline std::map<int, std::vector<amrex::ParticleReal>> h_vertices_y = {};
+
+        //! device: normal multipole coefficients of the magnetic field
+        inline std::map<int, amrex::Gpu::DeviceVector<amrex::ParticleReal>> d_vertices_x = {};
+        //! device: skew multipole coefficients of the magnetic field
+        inline std::map<int, amrex::Gpu::DeviceVector<amrex::ParticleReal>> d_vertices_y = {};
+
+    } // namespace PolygonApertureData
+
+    struct PolygonAperture
+    : public mixin::Named,
+      public mixin::BeamOptic<PolygonAperture>,
+      public mixin::LinearTransport<PolygonAperture>,
+      public mixin::Thin,
+      public mixin::Alignment,
+      public mixin::NoFinalize,
+      public amrex::simd::Vectorized<amrex::simd::native_simd_size_particlereal>
+    {
+        static constexpr auto type = "PolygonAperture";
+        using PType = ImpactXParticleContainer::ParticleType;
+
+        enum Action
+        {
+            transmit,
+            absorb
+        };
+
+        static std::string
+        action_name (Action const & action);
+
+        /** A thin collimator element that applies a transverse aperture boundary defined
+         *  by points of a polygon vertices.
+         *  Particles outside the boundary are considered lost.
+         *
+         * @param action specify action of domain (transmit/absorb)
+         * @param vertices_x array of coordinates of the horizontal vertex positions (m)
+         * @param vertices_y array of coordinates of the vertical vertex positions (m)
```

Comment:
```suggestion
         * @param vertices_xx array of coordinates of the horizontal vertex positions (m)
         * @param vertices_yy array of coordinates of the vertical vertex positions (m)
```
Doxygen doesn't like the fact that the two variable names given here are different from those that appear in the function argument list.